test_compare/cmp.py

```diff
@@ -0,0 +1,679 @@
+            case_cmp["percentage"] = None
+        cases.append(case_cmp)
+    for name, case in cases_new.items():
+        if name not in cases_old:
+            case_cmp = {}
+            case_cmp["name"] = name
+            case_cmp["time_old"] = None
+            case_cmp["status_old"] = "--"
+            case_cmp["time_new"] = case["time"]
+            case_cmp["status_new"] = case["status"]
+            case_cmp["difference"] = None
+            case_cmp["percentage"] = None
+            cases.append(case_cmp)
+    return cases
+
+
+def difference_and_percentage(old, new):
+    difference = new - old
+    percentage = 100 * difference / old if old > 0 else 0 if difference == 0 else INF
+    return difference, percentage
+
+
+def make_row(data, style, separator, prefix, args):
+    difference, percentage = difference_and_percentage(data["time_old"], data["time_new"])
+    color = CHANGE_COLORS[change_dir(difference, percentage, args)]
+    return Row(
+        style,
+        separator,
+        f"{prefix}{data['name']}",
+        f"{data['time_old']:.3f}",
+        f"{data['time_new']:.3f}",
+        color,
+        f"{difference:+.3f}",
+        f"{percentage:+.2f}" if percentage < 10000 else "-.--",
+    )
+
+
+def make_case_status_row(case):
+    return StatusRow(
+        "",
+        "┆",
+        f"  -{case['name']}",
+        case["status_old"],
+        case["status_new"],
+    )
+
+
+def make_case_row(case, args):
+    difference, percentage = difference_and_percentage(case["time_old"], case["time_new"])
+    color = CHANGE_COLORS[change_dir(difference, percentage, args)]
+    return Row(
+        "",
+        "┆",
+        f"  -{case['name']}",
+        f"{case['time_old']:.3f}",
+        f"{case['time_new']:.3f}",
+        color,
+        f"{difference:+.3f}",
+        f"{percentage:+.2f}" if percentage < 10000 else "-.--",
+    )
+
+
+@dataclass
+class Row:
+    style: str = ""
+    separator: str = ""
+    name: str = ""
+    time_old: str = ""
+    time_new: str = ""
+    color: str = ""
+    difference: str = ""
+    percentage: str = ""
+
+
+@dataclass
+class StatusRow:
+    style: str = ""
+    separator: str = ""
+    name: str = ""
+    status_old: str = ""
+    status_new: str = ""
+
+
+def print_rows(rows):
+    if not rows:
+        return
+    max_name_len = max(len(row.name) for row in rows)
+    max_name_len = max(max_name_len, 4)
+    for row in rows:
+        if row.name == "":
+            print()
+            continue
+        if row.name == "H":
+            print(f"{ESCAPE_BOLD}{'':>{max_name_len}}║{'OLD':^9}║{'NEW':^9}║{'DIFFERENCE':^20}{ESCAPE_RESET}")
+            print(
+                f"{ESCAPE_BOLD}{ESCAPE_UNDERLINE}{'NAME':^{max_name_len}}║"
+                f"{'TIME':^9}║{'TIME':^9}║{'TIME':^10}║{'PERCENTAGE':^9}{ESCAPE_RESET}"
+            )
+            continue
+        print(
+            f"{row.style}{row.name:<{max_name_len}}{ESCAPE_RESET}{row.separator}"
```

Comment:
most of this function should probably be done as `Row` class function. The `max_len_name` seems problematic as we need to scan all rows, but it could be an input parameter, maybe we would like to shorten the names regardless of the max len name (one test might have a really long name which would decrease the readability of all of the results.

I would appreciate adding second function to `Row` class - `to_csv()` and an option to output machine-readable results.

Not sure how we want to shorten max_name_len then